### PR TITLE
Refactors polynomial trend forecaster

### PR DESCRIPTION
**Pull request by thayeylolu**
Reference : https://github.com/alan-turing-institute/sktime/issues/955

We need to slowly re-factor all forecasters to be compliant with the new interface (see PR #912). i.e., no overrides of fit etc and descendant classes only implementing core logic, while the BaseForecaster implements check etc logic.

Currently this is running due to various downwards compatibility workarounds, which should be removed once all forecasters are ready (in "no-override" state)

The forecasters that still need a downstream refactor are:

  

- [x] PolynomialForecaster

### CODE REVIEW

File: sktime/forecasting/trend.py [hunk 1]
```diff
@@ -69,14 +74,11 @@ def fit(self, y, X=None, fh=None):
         -------
         self : returns an instance of self.
         """
-        self._is_fitted = False
 
         if X is not None:
```
Comment (fkiraly) [inline sktime/forecasting/trend.py:78]: while we're here, let's just remove this error raised, see #1014

**Review by fkiraly — CHANGES_REQUESTED** [review]
Looks great!

This is a full refactor and well done, but I think we can easily also address #1013 in this PR, by removing the superfluous error right with it.

File: sktime/forecasting/trend.py (hunk 1 above)
Comment (mloning) [reply sktime/forecasting/trend.py:78]: Agreed, probably legacy code before we had agreed on how to handle exogenous data.

Comment (thayeylolu) [reply sktime/forecasting/trend.py:78]: Hello @fkiraly , Is there something I am suppose to resolve here? I cannot seem to see it.

Comment (fkiraly) [reply sktime/forecasting/trend.py:78]: I meant, remove the `NotImplementedError` being thrown when `X` is passed. The problem is described in #1014.

This is not strictly part of the refactoring, but let's do this while we're already here.

Comment (thayeylolu) [reply sktime/forecasting/trend.py:78]: noted

**Review by fkiraly — APPROVED** [review]
looks ready to go, let's just wait for the checks to finish.

**Comment by thayeylolu** [comment]
@mloning this is ready for review :)

**Pull request merged**